server/user_account/views.py

```diff
@@ -6,14 +6,59 @@
 from rest_framework import status, viewsets, generics
 from rest_framework.response import Response
 from rest_framework.authtoken.models import Token
+from rest_framework.permissions import IsAuthenticated
 from .serializers import UserSerializer, LoginSerializer
 from .models import CustomUser
 
 
+
 class RegistrationView(viewsets.ModelViewSet):
 
     """
-    Creates a new System Admin in the db.
+    Creates a new user in the db.
+    """
+    permission_classes = [IsAuthenticated]
+    queryset = CustomUser.objects.all()
+    serializer_class = UserSerializer
+    http_method_names = ['post']
+
+    def create(self, request, *args, **kwargs):
+        """
+        :param request: request.data: first_name,
+        last_name, user_name, password, role, email, is_active
+        :return: user_name, token
+        """
+        # Retrieve the authenticated user making the request
+        auth_content = {
+        'user': str(request.user),
+        'auth': str(request.auth),
+        }
+        # If not authenticated return 401
+        if auth_content['auth'] is None:
```

Comment:
You don't need to add this check. This is handled by permission_classes=[IsAuthenticated]. So you can delete it. But just test it before in this particular case because we are overriding the method.

Reply:
That's true it is already handled by the permission class. I removed in dade84b836c4f152d1802526d2116771e057f97a